hypervisor/arch/x86/guest/vm.c

```diff
@@ -845,7 +845,7 @@ void update_vm_vlapic_state(struct acrn_vm *vm)
 	}
 
 	vm->arch_vm.vlapic_state = vlapic_state;
-	spinlock_release(&vm->vm_lock);
+	spinlock_release(&vm->vlapic_state_lock);
```

Comment:
Let us rename vlapic_state to vlapic_mode first, and then this lock to vlapic_mode_lock